docker-plugin/src/main/java/com/nirima/jenkins/plugins/docker/DockerComputerLauncher.java

```diff
@@ -46,15 +46,19 @@ private static SSHLauncher getSSHLauncher(InspectContainerResponse detail, Docke
 
             ExposedPort sshPort = new ExposedPort(22);
             int port = 22;
+            String host = null;
 
             Ports.Binding[] bindings = detail.getNetworkSettings().getPorts().getBindings().get(sshPort);
 
             for(Ports.Binding b : bindings) {
                 port = b.getHostPort();
+                host = b.getHostIp();
             }
 
-            URL hostUrl = new URL(template.getParent().serverUrl);
-            String host = hostUrl.getHost();
+            if (host == null) {
+                URL hostUrl = new URL(template.getParent().serverUrl);
+                host = hostUrl.getHost();
```

Comment:
Have no code under, hand afair some addresses already URL, so if it URL, then no need to create new objects. And usually URI is preferred instead URL.